xdsl/rewriter.py

```diff
@@ -186,3 +186,15 @@ def move_region_contents_to_new_regions(region: Region) -> Region:
             new_region.add_block(block)
         region.blocks = []
         return new_region
+
+    @staticmethod
+    def inline_region_before(region: Region, target: Block) -> None:
+        """Move the region blocks to an existing region."""
+        parent_region = target.parent
+        if parent_region is None:
```

Comment:
You also need to check that `target.parent` is not `region`, otherwise this will break.